sql: TestTelemetry failed

sql.TestTelemetry [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4159053&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4159053&tab=artifacts#/) on master @ [54c5a2c645756c8cbd3cf7dff1fd354c290b013f](https://github.com/cockroachdb/cockroach/commits/54c5a2c645756c8cbd3cf7dff1fd354c290b013f):


```
=== RUN   TestTelemetry
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTelemetry2553740697
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestTelemetry
    telemetry_test.go:45: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTelemetry2553740697
--- FAIL: TestTelemetry (56.18s)
=== RUN   TestTelemetry/planning/server
    telemetry.go:90: 
        testdata/telemetry/planning:191: SELECT x FROM (VALUES (1)) AS b(x) WHERE EXISTS(SELECT * FROM (VALUES (1)) AS a(x) WHERE a.x = b.x)
        expected:
        sql.plan.subquery.correlated
        
        found:
        sql.plan.subquery
        sql.plan.subquery.correlated
        --- FAIL: TestTelemetry/planning/server (2.11s)
=== RUN   TestTelemetry/planning
    --- FAIL: TestTelemetry/planning (3.50s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTelemetry.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql.TestTelemetry [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4161103&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4161103&tab=artifacts#/) on master @ [506412ffdb6d0f6d4952d2599b81cb9294a123a1](https://github.com/cockroachdb/cockroach/commits/506412ffdb6d0f6d4952d2599b81cb9294a123a1):


```
=== RUN   TestTelemetry
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTelemetry2017684981
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestTelemetry
    telemetry_test.go:45: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTelemetry2017684981
--- FAIL: TestTelemetry (70.18s)
=== RUN   TestTelemetry/planning/server
    telemetry.go:90: 
        testdata/telemetry/planning:191: SELECT x FROM (VALUES (1)) AS b(x) WHERE EXISTS(SELECT * FROM (VALUES (1)) AS a(x) WHERE a.x = b.x)
        expected:
        sql.plan.subquery.correlated
        
        found:
        sql.plan.subquery
        sql.plan.subquery.correlated
        --- FAIL: TestTelemetry/planning/server (2.15s)
=== RUN   TestTelemetry/planning
    --- FAIL: TestTelemetry/planning (3.93s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=deadlock

- GOFLAGS=-parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTelemetry.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/testdata/telemetry/planning
+++ b/pkg/sql/testdata/telemetry/planning
@@ -180,13 +180,17 @@ SELECT * FROM (VALUES (1), (2)) AS a(x) JOIN LATERAL (SELECT a.x+1 AS x) AS b ON
 sql.plan.lateral-join
 
 feature-allowlist
-sql.plan.subquery.*
+sql.plan.subquery
 ----
 
 feature-usage
 SELECT 1 = (SELECT a FROM x LIMIT 1)
 ----
 sql.plan.subquery
+
+feature-allowlist
+sql.plan.subquery.correlated
+----
 
 feature-usage
 SELECT x FROM (VALUES (1)) AS b(x) WHERE EXISTS(SELECT * FROM (VALUES (1)) AS a(x) WHERE a.x = b.x)

